mobile-academy/src/main/java/org/motechproject/nms/mobileacademy/service/impl/MobileAcademyServiceImpl.java

```diff
@@ -28,6 +36,10 @@
 @Service("mobileAcademyService")
 public class MobileAcademyServiceImpl implements MobileAcademyService {
 
+    private static final String COURSE_CONTENT_FILE = "nmsCourse.json";
+
+    private static final String COURSE_NAME = "MobileAcademyCourse";
+
     private static final String FINAL_BOOKMARK = "Chapter11_Quiz";
```

Comment:
There is a spec requirement ..but the other way. There is guaranteed to
only ever be 11 chapters and I believe the vxml parsing relies on this
'magic' number for their looping logic.

On Wed, May 27, 2015 at 3:30 PM, Lauren Lavoie <notifications@github.com>
wrote:

> In
> mobile-academy/src/main/java/org/motechproject/nms/mobileacademy/service/impl/MobileAcademyServiceImpl.java
> <https://github.com/motech-implementations/mim/pull/283#discussion_r31187533>
> :
>
> > @@ -28,6 +36,10 @@
> >  @Service("mobileAcademyService")
> >  public class MobileAcademyServiceImpl implements MobileAcademyService {
> >
> > +    private static final String COURSE_CONTENT_FILE = "nmsCourse.json";
> > +
> > +    private static final String COURSE_NAME = "MobileAcademyCourse";
> > +
> >      private static final String FINAL_BOOKMARK = "Chapter11_Quiz";
>
> Is there no spec requirement for these values to be configurable? It would
> be unfortunate if the course was changed to have 12 chapters and that
> required a recompile due to hard-coded constants...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/motech-implementations/mim/pull/283/files#r31187533>.
>